test/integ/test_cursor.py

```diff
@@ -1206,6 +1206,23 @@ def test_check_can_use_pandas(conn_cnx):
                 cur.check_can_use_pandas()
 
 
+@pytest.mark.skipolddriver
+def test_use_pandas_asynchronously(conn_cnx):
+    """Test get pandas in an asynchronous way"""
+
+    with conn_cnx() as cnx:
+        with cnx.cursor() as cur:
+            cur.execute("select 1/2")
+            res_sync = cur.fetch_pandas_batches()
+
+            result = cur.execute_async("select 1/2")
+            cur.get_results_from_sfqid(result["queryId"])
+            res_async = cur.fetch_pandas_batches()
+
+            for r_sync, r_async in zip(res_sync, res_async):
```

Comment:
if both res_sync and res_async are empty, no assertion will be performed, so I guess you need to check res_sync and res_async are not empty.